### ISSUE TITLE
Regex capture variables result in warnings

### ISSUE THREAD

**Issue by dcharleston** [fiddyspence/puppetlint-numericvariable#3]
Regex capture variables, described in https://docs.puppet.com/puppet/4.7/reference/lang_conditional.html#regex-capture-variables will trigger a warning.  Is there a way to have these special variables to be allowed?

**Comment by fiddyspence** [comment]
Yeah - off the top of my head, I'm not sure - that allowed behaviour isn't terribly generic, and I'm not sure whether the puppet lint parser is aware of the current scope of the token being evaluated.  I'll take a look, but at this point I'm not optimistic.
